srtcore/core.cpp

```diff
@@ -4234,7 +4273,10 @@ bool CUDT::processAsyncConnectRequest(EReadStatus         rst,
     {
         // m_RejectReason already set at worker_ProcessAddressedPacket.
         LOGC(cnlog.Warn,
-             log << "processAsyncConnectRequest: REJECT reported from HS processing, not processing further.");
+             log << "processAsyncConnectRequest: REJECT reported from HS processing:"
+             << srt_rejectreason_str(m_RejectReason)
+             << "- not processing further; REQ-TIME LOW");
+        m_tsLastReqTime = steady_clock::time_point();
```

Comment:
Resetting the last req time here causes that in case of password-related rejection the socket would be closed immediately and not processed further, which also means that the immediate state found on the socket by the application upon reception of epoll error would be now SRTS_BROKEN, instead of, like previously SRTS_CONNECTING.

This likely should be the fix for the first thing here, maybe instead it should be removed from here, but the process of rejection reviewed.